drivers/gpio/gpio_rp1.c

```diff
@@ -0,0 +1,222 @@
+		sys_clear_bits(PADS_CTRL(data->pads_base, pin),
+			       PADS_OUTPUT_DISABLE | PADS_INPUT_ENABLE);
+
+		if (flags & GPIO_OUTPUT_INIT_HIGH) {
+			sys_set_bit(RIO_OUT_SET(data->rio_base), pin);
+			sys_clear_bit(RIO_OUT_CLR(data->rio_base), pin);
+		} else if (flags & GPIO_OUTPUT_INIT_LOW) {
+			sys_set_bit(RIO_OUT_CLR(data->rio_base), pin);
+			sys_clear_bit(RIO_OUT_SET(data->rio_base), pin);
+		}
+	} else if (flags & GPIO_INPUT) {
+		sys_set_bit(RIO_OE_CLR(data->rio_base), pin);
+		sys_set_bits(PADS_CTRL(data->pads_base, pin),
+			     PADS_OUTPUT_DISABLE | PADS_INPUT_ENABLE);
+	}
+
+	/* Set pull up/down */
+	sys_clear_bits(PADS_CTRL(data->pads_base, pin),
+		       PADS_PULL_UP_ENABLE | PADS_PULL_DOWN_ENABLE);
+
+	if (flags & GPIO_PULL_UP) {
+		sys_set_bits(PADS_CTRL(data->pads_base, pin), PADS_PULL_UP_ENABLE);
+	} else if (flags & GPIO_PULL_DOWN) {
+		sys_set_bits(PADS_CTRL(data->pads_base, pin), PADS_PULL_DOWN_ENABLE);
+	}
+
+	return 0;
+}
+
+static int gpio_rp1_port_get_raw(const struct device *port, gpio_port_value_t *value)
+{
+	struct gpio_rp1_data *data = port->data;
+
+	*value = sys_read32(RIO_IN(data->rio_base));
+
+	return 0;
+}
+
+static int gpio_rp1_port_set_masked_raw(const struct device *port, gpio_port_pins_t mask,
+					gpio_port_value_t value)
+{
+	struct gpio_rp1_data *data = port->data;
+
+	sys_clear_bits(RIO_OUT_SET(data->rio_base), mask);
+	sys_set_bits(RIO_OUT_CLR(data->rio_base), mask);
+
+	sys_clear_bits(RIO_OUT_CLR(data->rio_base), (value & mask));
+	sys_set_bits(RIO_OUT_SET(data->rio_base), (value & mask));
+
+	return 0;
+}
+
+static int gpio_rp1_port_set_bits_raw(const struct device *port, gpio_port_pins_t pins)
+{
+	struct gpio_rp1_data *data = port->data;
+
+	sys_clear_bits(RIO_OUT_CLR(data->rio_base), pins);
+	sys_set_bits(RIO_OUT_SET(data->rio_base), pins);
+
+	return 0;
+}
+
+static int gpio_rp1_port_clear_bits_raw(const struct device *port, gpio_port_pins_t pins)
+{
+	struct gpio_rp1_data *data = port->data;
+
+	sys_clear_bits(RIO_OUT_SET(data->rio_base), pins);
+	sys_set_bits(RIO_OUT_CLR(data->rio_base), pins);
+
+	return 0;
+}
+
+static int gpio_rp1_port_toggle_bits(const struct device *port, gpio_port_pins_t pins)
+{
+	struct gpio_rp1_data *data = port->data;
+	uint32_t val;
+
+	val = sys_read32(RIO_OUT(data->rio_base));
+
+	/* Low to high */
+	sys_set_bits(RIO_OUT_SET(data->rio_base), val ^ pins);
+	sys_clear_bits(RIO_OUT_CLR(data->rio_base), val ^ pins);
+
+	/* High to low */
+	sys_set_bits(RIO_OUT_CLR(data->rio_base), val & pins);
+	sys_clear_bits(RIO_OUT_SET(data->rio_base), val & pins);
+
+	return 0;
+}
+
+static const struct gpio_driver_api gpio_rp1_api = {
+	.pin_configure = gpio_rp1_pin_configure,
+	.port_get_raw = gpio_rp1_port_get_raw,
+	.port_set_masked_raw = gpio_rp1_port_set_masked_raw,
+	.port_set_bits_raw = gpio_rp1_port_set_bits_raw,
+	.port_clear_bits_raw = gpio_rp1_port_clear_bits_raw,
+	.port_toggle_bits = gpio_rp1_port_toggle_bits,
+};
+
+int gpio_rp1_init(const struct device *port)
```

Comment:
Any reason why this can't be `static`?

```suggestion
static int gpio_rp1_init(const struct device *port)
```

Reply:
It seems I made a mistake. Thank you for pointing it out.